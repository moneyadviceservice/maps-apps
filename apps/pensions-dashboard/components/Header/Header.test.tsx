import { cleanup, fireEvent, render, screen } from '@testing-library/react';

import { Header } from '.';

import '@testing-library/jest-dom';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

describe('Header component', () => {
  beforeEach(() => {
    HTMLDialogElement.prototype.show = jest.fn();
    HTMLDialogElement.prototype.showModal = jest.fn();
    HTMLDialogElement.prototype.close = jest.fn();
    jest.clearAllMocks();
    cleanup();
  });

  it('renders correctly', () => {
    render(<Header />);
    const header = screen.getByTestId('header');
    expect(header).toMatchSnapshot();
  });

  it('does not render the log out option if not a logged in page excluded', () => {
    render(<Header isLoggedInPage={false} />);
    const logoutLink = screen.queryByTestId('log-out-link');
    expect(logoutLink).not.toBeInTheDocument();
  });

  it('renders correctly when the nav is opened with a click', () => {
    render(<Header />);
    const header = screen.getByTestId('header');
    const navToggle = screen.getByTestId('nav-toggle');
    fireEvent.click(navToggle);
    expect(header).toMatchSnapshot();
  });

  it('renders correctly when the nav is opened by a Space keypress', () => {
    render(<Header />);
    const header = screen.getByTestId('header');
    const navToggle = screen.getByTestId('nav-toggle');
    fireEvent.keyDown(navToggle, { key: ' ' });
    expect(header).toMatchSnapshot();
  });

  it('renders correctly when the nav is opened by an Enter keypress', () => {
    render(<Header />);
    const header = screen.getByTestId('header');
    const navToggle = screen.getByTestId('nav-toggle');
    fireEvent.keyDown(navToggle, { key: 'Enter' });
    expect(header).toMatchSnapshot();
  });

  it('fires handleLogout when log out link is clicked', () => {
    const handleLogout = jest.fn();
    render(<Header setIsLogoutModalOpen={handleLogout} />);
    const logoutLink = screen.getByTestId('logout-link');
    fireEvent.click(logoutLink);
    expect(handleLogout).toHaveBeenCalled();
  });

  it('fires handleLogout when log out link is activated by a keypress', () => {
    const handleLogout = jest.fn();
    render(<Header setIsLogoutModalOpen={handleLogout} />);
    const logoutLink = screen.getByTestId('logout-link');
    fireEvent.keyDown(logoutLink, { key: 'Enter' });
    expect(handleLogout).toHaveBeenCalled();
  });

  it('closes the modal when the cancel button is clicked', () => {
    render(<Header isLogoutModalOpen={true} />);
    const cancelButton = screen.getByTestId('logout-no');
    fireEvent.click(cancelButton);
    expect(cancelButton).toBeInTheDocument();
  });
});
