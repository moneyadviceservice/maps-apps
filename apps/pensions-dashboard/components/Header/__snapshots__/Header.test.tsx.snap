// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<header
  data-testid="header"
>
  <a
    class="sr-only focus:not-sr-only focus:bg-yellow-200 focus:z-20 focus:border-b-4 focus:border-b-blue-800 focus:absolute focus:p-2 focus:rounded focus:top-4 focus:left-4"
    href="#main"
  >
    Skip to content
  </a>
  <div
    class="relative z-10 flex items-center p-4 bg-blue-800 t-header print:hidden shadow-bottom-gray"
  >
    <details
      class="relative group"
    >
      <summary
        class="t-header-menu-open cursor-pointer list-none [&::-webkit-details-marker]:hidden text-center text-white text-sm"
        data-testid="nav-toggle"
        title="Open menu"
      >
        <div>
          <div
            class="group-open:hidden"
          >
            <svgr-mock
              classname=""
            />
            <span>
              menu
            </span>
          </div>
          <div
            class="hidden group-open:block"
          >
            <svgr-mock
              classname=""
            />
            <span>
              close
            </span>
          </div>
        </div>
      </summary>
      <nav
        class="absolute z-10 flex pt-4 text-gray-900 t-header-navigation w-80 md:w-96"
      >
        <div
          class="w-full pb-3 space-y-4 bg-white border"
        >
          <ul
            class="divide-y"
          >
            <li
              class="px-3 py-2"
            >
              <a
                class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
                href="/en"
              >
                <div
                  class="flex items-center"
                >
                  <div>
                    &lt;No translation available&gt;
                  </div>
                  <svg
                    aria-hidden="true"
                    height="32"
                    viewBox="0 0 24 24"
                    width="32"
                  >
                    <path
                      d="M8.59 16.58L13.17 12L8.59 7.41L10 6l6 6l-6 6l-1.41-1.42Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </a>
            </li>
          </ul>
          <div
            class="px-3 t-header-navigation-button"
          >
            <a
              class="border border-pink-600 text-pink-600 bg-white hover:border-pink-800 hover:text-pink-800 hover:shadow-none focus:border-transparent active:text-slate-900 active:outline-purple-700 active:bg-white active:border-transparent t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0"
              href="?language=cy"
              lang="cy"
            >
              Cymraeg
            </a>
          </div>
        </div>
      </nav>
    </details>
    <div
      class="mx-auto t-header-logo"
    >
      <span
        class="hidden md:inline"
      >
        <svgr-mock
          classname=""
        />
      </span>
      <span
        class="md:hidden"
      >
        <svgr-mock
          classname=""
        />
      </span>
    </div>
    <nav
      aria-label="Language switcher"
      class="hidden md:block absolute right-4"
    >
      <ul>
        <li
          class="inline-block after:content-[''] after:inline-block after:h-6 after:relative after:top-2 after:border-white after:border-r-1"
        >
          <span
            aria-current="true"
            class="p-2 px-4 text-white"
          >
            English
          </span>
        </li>
        <li
          class="inline-block"
        >
          <a
            class="group p-2 px-4 inline-flex items-center gap-2 text-white no-underline hover:text-pink-400 visited:text-white focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 hover:underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
            href="?language=cy"
            hreflang="cy"
            lang="cy"
            rel="alternate"
          >
            <span
              class="sr-only"
            >
              Newid yr iaith ir Gymraeg
            </span>
            <span
              aria-hidden="true"
              class="inline-block relative after:content-[''] after:inline-block after:absolute after:-bottom-1 after:left-0 after:bg-white after:w-full after:h-[1px] group-hover:after:h-[2px] "
            >
              Cymraeg
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;

exports[`Header component renders correctly when the nav is opened by a Space keypress 1`] = `
<header
  data-testid="header"
>
  <a
    class="sr-only focus:not-sr-only focus:bg-yellow-200 focus:z-20 focus:border-b-4 focus:border-b-blue-800 focus:absolute focus:p-2 focus:rounded focus:top-4 focus:left-4"
    href="#main"
  >
    Skip to content
  </a>
  <div
    class="fixed top-0 bottom-0 left-0 right-0 z-10 bg-black opacity-25 t-menu-overlay"
  />
  <div
    class="relative z-10 flex items-center p-4 bg-blue-800 t-header print:hidden shadow-bottom-gray"
  >
    <details
      class="relative group"
      open=""
    >
      <summary
        class="t-header-menu-close cursor-pointer list-none [&::-webkit-details-marker]:hidden text-center text-white text-sm"
        data-testid="nav-toggle"
        title="Close menu"
      >
        <div>
          <div
            class="group-open:hidden"
          >
            <svgr-mock
              classname=""
            />
            <span>
              menu
            </span>
          </div>
          <div
            class="hidden group-open:block"
          >
            <svgr-mock
              classname=""
            />
            <span>
              close
            </span>
          </div>
        </div>
      </summary>
      <nav
        class="absolute z-10 flex pt-4 text-gray-900 t-header-navigation w-80 md:w-96"
      >
        <div
          class="w-full pb-3 space-y-4 bg-white border"
        >
          <ul
            class="divide-y"
          >
            <li
              class="px-3 py-2"
            >
              <a
                class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
                href="/en"
              >
                <div
                  class="flex items-center"
                >
                  <div>
                    &lt;No translation available&gt;
                  </div>
                  <svg
                    aria-hidden="true"
                    height="32"
                    viewBox="0 0 24 24"
                    width="32"
                  >
                    <path
                      d="M8.59 16.58L13.17 12L8.59 7.41L10 6l6 6l-6 6l-1.41-1.42Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </a>
            </li>
          </ul>
          <div
            class="px-3 t-header-navigation-button"
          >
            <a
              class="border border-pink-600 text-pink-600 bg-white hover:border-pink-800 hover:text-pink-800 hover:shadow-none focus:border-transparent active:text-slate-900 active:outline-purple-700 active:bg-white active:border-transparent t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0"
              href="?language=cy"
              lang="cy"
            >
              Cymraeg
            </a>
          </div>
        </div>
      </nav>
    </details>
    <div
      class="mx-auto t-header-logo"
    >
      <span
        class="hidden md:inline"
      >
        <svgr-mock
          classname=""
        />
      </span>
      <span
        class="md:hidden"
      >
        <svgr-mock
          classname=""
        />
      </span>
    </div>
    <nav
      aria-label="Language switcher"
      class="hidden md:block absolute right-4"
    >
      <ul>
        <li
          class="inline-block after:content-[''] after:inline-block after:h-6 after:relative after:top-2 after:border-white after:border-r-1"
        >
          <span
            aria-current="true"
            class="p-2 px-4 text-white"
          >
            English
          </span>
        </li>
        <li
          class="inline-block"
        >
          <a
            class="group p-2 px-4 inline-flex items-center gap-2 text-white no-underline hover:text-pink-400 visited:text-white focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 hover:underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
            href="?language=cy"
            hreflang="cy"
            lang="cy"
            rel="alternate"
          >
            <span
              class="sr-only"
            >
              Newid yr iaith ir Gymraeg
            </span>
            <span
              aria-hidden="true"
              class="inline-block relative after:content-[''] after:inline-block after:absolute after:-bottom-1 after:left-0 after:bg-white after:w-full after:h-[1px] group-hover:after:h-[2px] "
            >
              Cymraeg
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;

exports[`Header component renders correctly when the nav is opened by an Enter keypress 1`] = `
<header
  data-testid="header"
>
  <a
    class="sr-only focus:not-sr-only focus:bg-yellow-200 focus:z-20 focus:border-b-4 focus:border-b-blue-800 focus:absolute focus:p-2 focus:rounded focus:top-4 focus:left-4"
    href="#main"
  >
    Skip to content
  </a>
  <div
    class="fixed top-0 bottom-0 left-0 right-0 z-10 bg-black opacity-25 t-menu-overlay"
  />
  <div
    class="relative z-10 flex items-center p-4 bg-blue-800 t-header print:hidden shadow-bottom-gray"
  >
    <details
      class="relative group"
      open=""
    >
      <summary
        class="t-header-menu-close cursor-pointer list-none [&::-webkit-details-marker]:hidden text-center text-white text-sm"
        data-testid="nav-toggle"
        title="Close menu"
      >
        <div>
          <div
            class="group-open:hidden"
          >
            <svgr-mock
              classname=""
            />
            <span>
              menu
            </span>
          </div>
          <div
            class="hidden group-open:block"
          >
            <svgr-mock
              classname=""
            />
            <span>
              close
            </span>
          </div>
        </div>
      </summary>
      <nav
        class="absolute z-10 flex pt-4 text-gray-900 t-header-navigation w-80 md:w-96"
      >
        <div
          class="w-full pb-3 space-y-4 bg-white border"
        >
          <ul
            class="divide-y"
          >
            <li
              class="px-3 py-2"
            >
              <a
                class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
                href="/en"
              >
                <div
                  class="flex items-center"
                >
                  <div>
                    &lt;No translation available&gt;
                  </div>
                  <svg
                    aria-hidden="true"
                    height="32"
                    viewBox="0 0 24 24"
                    width="32"
                  >
                    <path
                      d="M8.59 16.58L13.17 12L8.59 7.41L10 6l6 6l-6 6l-1.41-1.42Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </a>
            </li>
          </ul>
          <div
            class="px-3 t-header-navigation-button"
          >
            <a
              class="border border-pink-600 text-pink-600 bg-white hover:border-pink-800 hover:text-pink-800 hover:shadow-none focus:border-transparent active:text-slate-900 active:outline-purple-700 active:bg-white active:border-transparent t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0"
              href="?language=cy"
              lang="cy"
            >
              Cymraeg
            </a>
          </div>
        </div>
      </nav>
    </details>
    <div
      class="mx-auto t-header-logo"
    >
      <span
        class="hidden md:inline"
      >
        <svgr-mock
          classname=""
        />
      </span>
      <span
        class="md:hidden"
      >
        <svgr-mock
          classname=""
        />
      </span>
    </div>
    <nav
      aria-label="Language switcher"
      class="hidden md:block absolute right-4"
    >
      <ul>
        <li
          class="inline-block after:content-[''] after:inline-block after:h-6 after:relative after:top-2 after:border-white after:border-r-1"
        >
          <span
            aria-current="true"
            class="p-2 px-4 text-white"
          >
            English
          </span>
        </li>
        <li
          class="inline-block"
        >
          <a
            class="group p-2 px-4 inline-flex items-center gap-2 text-white no-underline hover:text-pink-400 visited:text-white focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 hover:underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
            href="?language=cy"
            hreflang="cy"
            lang="cy"
            rel="alternate"
          >
            <span
              class="sr-only"
            >
              Newid yr iaith ir Gymraeg
            </span>
            <span
              aria-hidden="true"
              class="inline-block relative after:content-[''] after:inline-block after:absolute after:-bottom-1 after:left-0 after:bg-white after:w-full after:h-[1px] group-hover:after:h-[2px] "
            >
              Cymraeg
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;

exports[`Header component renders correctly when the nav is opened with a click 1`] = `
<header
  data-testid="header"
>
  <a
    class="sr-only focus:not-sr-only focus:bg-yellow-200 focus:z-20 focus:border-b-4 focus:border-b-blue-800 focus:absolute focus:p-2 focus:rounded focus:top-4 focus:left-4"
    href="#main"
  >
    Skip to content
  </a>
  <div
    class="fixed top-0 bottom-0 left-0 right-0 z-10 bg-black opacity-25 t-menu-overlay"
  />
  <div
    class="relative z-10 flex items-center p-4 bg-blue-800 t-header print:hidden shadow-bottom-gray"
  >
    <details
      class="relative group"
      open=""
    >
      <summary
        class="t-header-menu-close cursor-pointer list-none [&::-webkit-details-marker]:hidden text-center text-white text-sm"
        data-testid="nav-toggle"
        title="Close menu"
      >
        <div>
          <div
            class="group-open:hidden"
          >
            <svgr-mock
              classname=""
            />
            <span>
              menu
            </span>
          </div>
          <div
            class="hidden group-open:block"
          >
            <svgr-mock
              classname=""
            />
            <span>
              close
            </span>
          </div>
        </div>
      </summary>
      <nav
        class="absolute z-10 flex pt-4 text-gray-900 t-header-navigation w-80 md:w-96"
      >
        <div
          class="w-full pb-3 space-y-4 bg-white border"
        >
          <ul
            class="divide-y"
          >
            <li
              class="px-3 py-2"
            >
              <a
                class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
                href="/en"
              >
                <div
                  class="flex items-center"
                >
                  <div>
                    &lt;No translation available&gt;
                  </div>
                  <svg
                    aria-hidden="true"
                    height="32"
                    viewBox="0 0 24 24"
                    width="32"
                  >
                    <path
                      d="M8.59 16.58L13.17 12L8.59 7.41L10 6l6 6l-6 6l-1.41-1.42Z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </a>
            </li>
          </ul>
          <div
            class="px-3 t-header-navigation-button"
          >
            <a
              class="border border-pink-600 text-pink-600 bg-white hover:border-pink-800 hover:text-pink-800 hover:shadow-none focus:border-transparent active:text-slate-900 active:outline-purple-700 active:bg-white active:border-transparent t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0"
              href="?language=cy"
              lang="cy"
            >
              Cymraeg
            </a>
          </div>
        </div>
      </nav>
    </details>
    <div
      class="mx-auto t-header-logo"
    >
      <span
        class="hidden md:inline"
      >
        <svgr-mock
          classname=""
        />
      </span>
      <span
        class="md:hidden"
      >
        <svgr-mock
          classname=""
        />
      </span>
    </div>
    <nav
      aria-label="Language switcher"
      class="hidden md:block absolute right-4"
    >
      <ul>
        <li
          class="inline-block after:content-[''] after:inline-block after:h-6 after:relative after:top-2 after:border-white after:border-r-1"
        >
          <span
            aria-current="true"
            class="p-2 px-4 text-white"
          >
            English
          </span>
        </li>
        <li
          class="inline-block"
        >
          <a
            class="group p-2 px-4 inline-flex items-center gap-2 text-white no-underline hover:text-pink-400 visited:text-white focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 hover:underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
            href="?language=cy"
            hreflang="cy"
            lang="cy"
            rel="alternate"
          >
            <span
              class="sr-only"
            >
              Newid yr iaith ir Gymraeg
            </span>
            <span
              aria-hidden="true"
              class="inline-block relative after:content-[''] after:inline-block after:absolute after:-bottom-1 after:left-0 after:bg-white after:w-full after:h-[1px] group-hover:after:h-[2px] "
            >
              Cymraeg
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
